test/modules/obsagent-client/src/main.rs

```diff
@@ -0,0 +1,204 @@
+use futures::stream::Stream;
+use futures::StreamExt;
+// use opentelemetry::global::shutdown_tracer_provider;
+use opentelemetry::sdk::metrics::{selectors, PushController};
+// use opentelemetry::trace::TraceError;
+use opentelemetry::global;
+use opentelemetry::metrics::{self, ObserverResult, ValueRecorder};
+use opentelemetry_otlp::ExporterConfig;
+use std::error::Error;
+use std::time::Duration;
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
+
+use clap::{value_t, App, Arg};
+use rand::Rng;
+
+// Skip first immediate tick from tokio, not needed for async_std.
+fn delayed_interval(duration: Duration) -> impl Stream<Item = tokio::time::Instant> {
+    opentelemetry::util::tokio_interval_stream(duration).skip(1)
+}
```

Comment:
This is in the tutorial we both came across:
https://github.com/open-telemetry/opentelemetry-rust/blob/main/examples/basic/src/main.rs#L25

Not sure what it is doing though. Do you?

Reply:
I'm not completely sure because it's not documented in the opentelemetry-otlp docs. I just know it's passed into `opentelemetry_otlp::new_metrics_pipeline()` and gets called with the default period (10 seconds) or passed in `period` so that metrics are pushed/exported from the pipeline every `period` seconds. It's not clear to me why the `interval` argument is needed and why the `period` value alone is not enough.  I'll ask this question in the opentelemetry-rust gitter channel and circle back to you when I get an answer.

Reply:
It looks like the purpose of the interval parameter is to support all async runtimes in a generic way so that for example people using async-std instead of tokio can specify a different interval function.